integration-tests/smoke/ccip/ccip_reorg_test.go

```diff
@@ -0,0 +1,675 @@
+		require.GreaterOrEqual(t, len(chain.HTTPRPCs), 1)
+		details, err := chainsel.GetChainDetailsByChainIDAndFamily(strconv.FormatUint(chain.ChainID, 10), chainsel.FamilyEVM)
+		require.NoError(t, err)
+
+		chainSelToRPCURL[details.ChainSelector] = chain.HTTPRPCs[0].Internal
+	}
+
+	reorgDestClient := ctf_client.NewRPCClient(chainSelToRPCURL[destChainSelector], nil)
+
+	// setup lanes from s1 and s2 to destChainSelector
+	testhelpers.AddLaneWithDefaultPricesAndFeeQuoterConfig(t, &e, state, sourceSelector, destChainSelector, false)
+
+	// wait for log poller filters to get registered.
+	l.Info().Msg("waiting for log poller filters to get registered")
+	time.Sleep(15 * time.Second)
+	reorgingMsgEvent := testhelpers.TestSendRequest(t, e.Env, state, sourceSelector, destChainSelector, false, router.ClientEVM2AnyMessage{
+		Receiver:     common.LeftPadBytes(state.Chains[destChainSelector].Receiver.Address().Bytes(), 32),
+		Data:         []byte("hello world"),
+		TokenAmounts: nil,
+		FeeToken:     common.HexToAddress("0x0"),
+		ExtraArgs:    nil,
+	})
+	l.Info().Msgf("sent CCIP message that will get re-orged, msg id: %x", reorgingMsgEvent.Message.Header.MessageId)
+
+	// Run reorg above finality depth
+	const reorgDepth = 60
+	l.Info().
+		Int("reorgDepth", reorgDepth).
+		Uint64("destChainSelector", destChainSelector).
+		Msg("starting blockchain reorg on Simulated Geth chain")
+	err = reorgDestClient.GethSetHead(reorgDepth)
+	require.NoError(t, err, "error starting blockchain reorg on Simulated Geth chain")
+
+	nodeAPIs := dockerEnv.GetCLClusterTestEnv().ClCluster.NodeAPIs()
+	nonBootstrapCount := len(nodeAPIs) - 1
+	l.Info().Msgf("waiting for %d non-bootstrap nodes to report finality violation on the logpoller", nonBootstrapCount)
+	require.Eventually(t, func() bool {
+		violatedResponses := make(map[string]struct{})
+		for _, node := range nodeAPIs {
+			// skip bootstrap nodes, they won't have any logpoller filters
+			p2pKeys, err := node.MustReadP2PKeys()
+			require.NoError(t, err)
+
+			l.Debug().Msgf("got p2pKeys from node API: %+v", p2pKeys)
+
+			require.GreaterOrEqual(t, len(p2pKeys.Data), 1)
+			if !slices.Contains(nonBootstrapP2PIDs, p2pKeys.Data[0].Attributes.PeerID) {
+				l.Info().Msgf("skipping bootstrap node w/ p2p id %s", p2pKeys.Data[0].Attributes.PeerID)
+				continue
+			}
+
+			resp, _, err := node.Health()
+			require.NoError(t, err)
+			for _, d := range resp.Data {
+				if d.Attributes.Name == destLogPollerService &&
+					d.Attributes.Output == "finality violated" &&
+					d.Attributes.Status == "failing" {
+					violatedResponses[p2pKeys.Data[0].Attributes.PeerID] = struct{}{}
+					break
+				}
+			}
+
+			if _, ok := violatedResponses[p2pKeys.Data[0].Attributes.PeerID]; ok {
+				l.Info().Msgf("node %s reported finality violation", p2pKeys.Data[0].Attributes.PeerID)
+			} else {
+				l.Info().Msgf("node %s did not report finality violation, log poller response: %+v",
```

Comment:
nit: a wrapper func for checking the finality violation is reported and not reported given the chain details